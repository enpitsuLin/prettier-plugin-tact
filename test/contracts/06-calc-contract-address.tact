import "@stdlib/deploy";

// first contract
contract Todo1 with Deployable {

    seqno: Int as uint64 = 1; // the code specifies the index (sequence number)

    init() {}

    get fun myAddress(): Address {
        return myAddress();
    }

    get fun otherAddress(): Address {
        let init: StateInit = initOf Todo2();
        return contractAddress(init);
    }
}

// second contract
contract Todo2 with Deployable {

    seqno: Int as uint64 = 2; // the code specifies the index (sequence number)

    init() {}

    get fun myAddress(): Address {
        return myAddress();
    }

    get fun otherAddress(): Address {
        let init: StateInit = initOf Todo1();
        return contractAddress(init);
    }
}