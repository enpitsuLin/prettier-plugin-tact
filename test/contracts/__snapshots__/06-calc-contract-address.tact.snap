import "@stdlib/deploy";

// first contract
contract Todo1 with Deployable {
    seqno: Int as uint64;// the code specifies the index (sequence number)


    get fun myAddress(): Address {
        return ;
    }

    get fun otherAddress(): Address {
        let init: StateInit = ;
        return ;
    }
}

// second contract
contract Todo2 with Deployable {
    seqno: Int as uint64;// the code specifies the index (sequence number)


    get fun myAddress(): Address {
        return ;
    }

    get fun otherAddress(): Address {
        let init: StateInit = ;
        return ;
    }
}